initial commit to enable Victara support

angelpipe opened the pull request:
Hey,
I'm intending to enable Victara(Moto X 2nd generation) support. I have given some steps but i want to make sure I'm pointing in the right direction, and I still have some doubts. Please forgive me if i do very basic questions (I didn't know anything about AOSP building a week ago).
## The approach

I want to use the most i can from AOSP because porting is easier as your current patches are written for it and mako board is a good starting point. Taking that in account, I'm using the same android-5.1.1_r1.xml file in my custom victara.xml. After that, I'm adding Cyanogen Mod's Victara repo into /device/motorola/victara; and finally, I add motorola vendor files into /vendor/motorola. Am I crazy or this should be enough to get an almost pure AOSP build for the device?. I hope I'm on the right path, you can check the details of this in the victara.xml file.
## Patches

As i used the same AOSP version you're using for mako, i guess the only patches that i need to review are the device ones (is that assumption ok?). I found 5 patches that are candidates to be ported to victara's build. I have rewritten some and i have doubts in others, so i will give a status for each one of them.
- **0001-Add-silk-product.patch** _PORTED_
  This patch is currently ported in this branch. If you find a problem about it, feel free to point it out.
- **0001-Don-t-enable-MTP-by-default.patch** _NOT PORTED_
  I haven't found any reference on CM's code where _persist.sys.usb.config's_ value is set to mtp. Does it mean this patch is not necessary?. The file where the var is referenced is https://github.com/CyanogenMod/android_device_motorola_victara/blob/cm-12.1/rootdir/etc/init.mmi.usb.sh
- **0001-Enable-firmware-to-mount-with-SELinux-disabled.patch** _NOT PORTED_
  In the fstab file from CM(https://github.com/CyanogenMod/android_device_motorola_victara/blob/cm-12.1/rootdir/etc/fstab.qcom), the /dev/block/platform/msm_sdcc.1/by-name/modem and /dev/block/platform/msm_sdcc.1/by-name/persist volumes are not listed so it is not possible to apply the change for the modem one. Should I add those volumes? could it affect the build?.
- **0001-Keep-2-cores-online-at-1.2ghz-to-stay-cool.patch** _NOT PORTED_
  I have a big question here, Does Silk use only 2 underclocked cores all the time?. Besides that, i think this is the trickiest patch because this kind of configuration for mako is done only in _init.mako.rc_; but in CM it is done in two files https://github.com/CyanogenMod/android_device_motorola_victara/blob/cm-12.1/rootdir/etc/init.target.rc and https://github.com/CyanogenMod/android_device_motorola_victara/blob/cm-12.1/rootdir/etc/init.qcom.rc. I don't know in which one of those should I set the patch, my guess is _init.qcom.rc_ as it includes _init.target.rc_. The other thing is I'm wondering if this patch is really necessary because CM does a lot of cpu settings in these files (could this patch as it is on mako cause conflicts with current CM settings?).
- **0001-sdcard-remove-derived-permissions.patch** _PORTED_
  This patch is currently ported in this branch. If you find a problem about it, feel free to point it out.

Thank you so much for reading this long text!

Best Regards.

benturner committed:
Upgrade to latest bleno

philikon committed:
Uniquify package names

The React Native packager is pretty picky about not having dupes in the entire tree, which also includes our Flow interface files. The latter should just be called `.js.flow` from now on.

Commit:
Add silk-movie

lightsofapollo committed:
Use new paths for sdk-emulator

Commit:
Update docs

Commit:
Add splash screen

Commit:
Add silk-vibrator

Commit:
Add silk-ntp

Commit:
Add silk-input

Commit:
Remove unneeded dependencies

Commit:
update the docs

Commit:
Make the template more developer-friendly*

Commit:
Doc nits

Commit:
Set CI_IGNORE_CASE_INSENSITIVE_FS

Commit:
Merge pull request #1660 from philikon/flow025

automerge

Commit:
force for Darwin

lightsofapollo committed:
sdk: uprade which to 1.2.8

Commit:
Merge pull request #1651 from mvines/splash

SOS vibrator, movie, ntp, input APIs

Commit:
Add silk-core-version

lightsofapollo committed:
Allow modules build via android to avoid npm install.

Introduces support for package.json.scripts.silk-build-aosp which will be
used in place of running the full npm install path (for in tree
node_modules).

Commit:
Clean up includes

Commit:
Merge pull request #1672 from mvines/inc

Clean up includes

Commit:
Merge pull request #1667 from mvines/splash

Add silk-core-version

Commit:
Open source silk-lights module closes #1652

Commit:
Add silk-capture

Commit:
Deprop

benturner committed:
Split bluetooth into its own child process

Commit:
Update documentation for silk-lights

Commit:
Add silk-camera

jainanshul committed:
Merge pull request #1677 from anefzaoui/docs-update-1

update the docs

Commit:
Open source silk-sensors closes #1653

Commit:
Deprop

Commit:
Update documentation for silk-sensors

Commit:
Merge pull request #1682 from jainanshul/open_source

automerge

Commit:
0.16.0

Commit:
Merge pull request #1683 from jainanshul/silk-sensors

automerge

Commit:
Remove my_inplace_build_module

Commit:
Merge pull request #1640 from lightsofapollo/fix-sos-paths

Fix sos paths

Commit:
Check for TRAVIS_BRANCH

Commit:
0.16.0

Commit:
Display the download URL

Commit:
Merge pull request #1681 from mvines/rg

automerge

Commit:
Merge pull request #1690 from mvines/inc

Inc

Commit:
Break up install/script steps

Commit:
Remove unused projects

Commit:
Don't bother building gtest/strace tests

Commit:
Add support for |./sync [product]|

Commit:
Merge pull request #1698 from mvines/prune

automerge

Commit:
Merge pull request #1699 from mvines/sync

automerge

Commit:
Permit folder modules to build into .silkslug/ only

lightsofapollo committed:
buildjs: Upgrade to allow building native modules

Commit:
Add sqlite for otapackage

Commit:
Tweak silk-alog docs

Commit:
Add Kenzo audio settings

Commit:
Docs overall updates

lightsofapollo committed:
add package.json to silk-init

lightsofapollo committed:
Misc changes to SoS for in tree node_modules

jainanshul committed:
Merge pull request #1724 from anefzaoui/docs-update-2

Docs overall updates

Commit:
Second iteration of doc updates

Commit:
Free up disk space before OTA generation

lightsofapollo committed:
SoS: Add package.json

lightsofapollo committed:
SoS: Check in node_modules

lightsofapollo committed:
sos: Update modules for in tree node_modules folder

lightsofapollo committed:
sos: Update various modules to latest android node module glue

lightsofapollo committed:
Merge pull request #1665 from lightsofapollo/land-node-modules

New cli tools for replacing npm and managing node_modules in tree.

lightsofapollo committed:
cli: Fix flow

philikon committed:
Don't poison my lunch with node 6.2 and npm 3.8

lightsofapollo committed:
version: Add runtime variant of version

philikon committed:
Bump babel-eslint peerDependency to correct version

lightsofapollo committed:
Merge pull request #1753 from lightsofapollo/runtime-silk-version

Runtime silk version

lightsofapollo committed:
version: Use dist/index.json as main for browser

Commit:
Fix license

Commit:
BUILD_FIRGERPRINT

jainanshul committed:
Fix invalid version number

Fall out from #18b5e126c3928fbe99c713e58ee1722252ac6abf

Commit:
Update gitignores

lightsofapollo committed:
Update node_module.mk

Commit:
Switch to a blob kernel for Kenzo

A very sad day.  The currently known Kenzo kernel source has non-functional
audio and camera so it's unusable.  C'est la guerre.

lightsofapollo committed:
buildjs: More docs

lightsofapollo committed:
Merge pull request #1780 from silklabs/lightsofapollo-patch-1

Update node_module.mk

Commit:
Set camera focus mode for auto on Kenzo

The default of 'continuous-video' causes Kenzo to move to manual focus with a
focal length of roughly 10cm...

Commit:
Fix buildjs symlinks

Commit:
Add silk-bledroid to PRODUCT_PACKAGES

Commit:
Merge pull request #1809 from mvines/focus

automerge

Commit:
Add missing internal node_module links

Commit:
Add silk-build-gonk script, for integration with the gonk build system

Commit:
Merge pull request #1817 from mvines/link

automerge

Commit:
Dedup network scan results.

Only the strongest bssid for each ssid is now reported.

Commit:
Merge pull request #1820 from mvines/sos

automerge

benturner committed:
Fix useless churn in mfg data

benturner committed:
Merge pull request #1821 from benturner/profiling

Setup issues revealed by profiling

lightsofapollo committed:
patch build: Avoid infinitely traversing previoiusly seen symlinks

Commit:
|silk activate --device xxxx ...| now works

Commit:
Merge pull request #1829 from lightsofapollo/dont-get-confused-on-symlink

automerge

Commit:
Merge pull request #1832 from mvines/act

`silk activate --device xxxx ...` now works

Commit:
Mark kissfft as external for webpack

benturner committed:
Instrument bledroid for systrace

Commit:
Merge pull request #1842 from benturner/profiling

automerge

Commit:
Capture video encoding can now be disabled by ro.silk.camera.video

Commit:
Merge pull request #1841 from jainanshul/in_tree_kissfft

automerge

Commit:
Remove silk-capture preinstall

Commit:
Add node_modules/opencv

Commit:
Remove silk-cv preinstall

Commit:
Tell node-gyp to use all cores

Commit:
Add node_modules/mocha

Commit:
Fix up log tests

Commit:
Add node-opencv gitignore

Commit:
Unbreak API doc generation

Commit:
Update logo

Commit:
Flash splash.img if present

Commit:
Add kenzo splash parition

philikon committed:
Upgrade to Flow 0.26

philikon committed:
One .flowconfig to rule them all

philikon committed:
Harmonize .flowconfigs in the tree

First step towards a global Flow run

Commit:
Reuse ccache between CI builds

Commit:
Add node-hid to gonkjs

Commit:
Support BTN_TOUCH key

Commit:
Merge pull request #1864 from philikon/harmonize_flow

automerge

Commit:
Eradicate LOCAL_NODE_MODULE_TYPE

Commit:
Merge pull request #1873 from philikon/uniflow

automerge

philikon committed:
Actually make silk-movie.js.flow a Flow interface file

philikon committed:
Small tweaks to docs/package.json

philikon committed:
HAXX: get documentationjs to parse Flow interface files

Commit:
0.17.0

Commit:
Add postsync hook for fetching non-repo managed things

philikon committed:
Get rid of unnecessary symlinks, RN packager doesn't like them

Commit:
Simplify libpreview loading

Commit:
libpreview now supports multiple clients

Commit:
Merge pull request #1850 from mvines/1727

Capture video encoding can now be disabled by ro.silk.camera.video

philikon committed:
Prefix package names with silk-

Commit:
Merge pull request #1890 from mvines/lp

Simplify libpreview loading

Commit:
Revert "Simplify libpreview loading"

This reverts commit d0c2854db4d01f9c48eb191b68d89535dff31727.

Commit:
buildjs_dir fixup

Commit:
Emit "fast frame" along side normal frames too

Commit:
Exit program on uncaught/unhandled errors

Commit:
Add more default log tags

Commit:
Throw on too many arguments

Commit:
Demote ignored message to debug

Commit:
Re-throw errors emitted in event callbacks cleanly

Commit:
0.17.0

Commit:
Only emit preview frames if there are listeners

philikon committed:
Bunch up all the trivial 'npm test' Travis runs in one big run

It's tedious to create boilerplate .travis.yml files for each new package, and it's easy to forget. At least one npm package in the tree (silk-rtc) didn't have one, for instance. This makes the process much easier.

lightsofapollo committed:
event-to-promise: 0.7.0

Commit:
Avoid killing capture process when ro.silk.camera.hw.enabled=true

Commit:
intel-tbb/intel-tbb just vanished, use local clone

Commit:
Mark silk-caffe as external in webpack config

Commit:
Disable gc tracing by default, too chatty

Commit:
Favour /data/node_modules in debug builds to keep /system pristine

Commit:
Merge pull request #1921 from mvines/dev

automerge

Commit:
Merge pull request #1934 from mvines/data

automerge

Commit:
Emit silk.module.push system prop once a module push completes

Commit:
Add option to push modules to /system

Commit:
Set default |silk build| output to .silkslug rather than .

Commit:
Use SILK_OUT for SILK_BUILDJS_DEST if present

Commit:
Add node_modules/.bin to silk build env

Commit:
Power Management in Silk template

Commit:
Include libpreview (#1955)

Commit:
Merge pull request #1946 from mvines/reload

Generalize device node_module reloading support

Commit:
Move ext.js module reloading into silk-modwatcher

Commit:
Add PropWatcher

Commit:
Use CC/CXX env variables to override gcc/g++ to get osx ci happy

Commit:
Merge pull request #1899 from mvines/lp

libpreview now supports multiple clients

Commit:
Merge pull request #1912 from mvines/sdk

automerge

Commit:
Merge pull request #1961 from mvines/osx

automerge

Commit:
Prune NODE_PATH

Commit:
Add symlink dependencies

Commit:
Remove postinstall

Commit:
Freshen up package.json

Commit:
Correct example

Commit:
Lint!

Commit:
Ensure gnu.sh exists in published package

Commit:
webpack is a peer

Commit:
Use carrot

Commit:
Mark node-hid as external

lightsofapollo committed:
buildjs: rewrite external path requires fixes #1973

lightsofapollo committed:
buildjs: Allow passing --no-native to disable building native modules

lightsofapollo committed:
buildjs: Allow disabling babel cache

Commit:
Merge pull request #1954 from anefzaoui/battery-management

Power Management in Silk template

Commit:
0.17.1

Commit:
Add silk-netstatus

Commit:
Mark _throwyEmit private

Commit:
Fix comment

Commit:
Add silk-volume

lightsofapollo committed:
open source babel-run

lightsofapollo committed:
roll babel-register into babel-run to make it standalone

Commit:
Merge pull request #1992 from lightsofapollo/pub-babel-run

automerge

Commit:
Update webpack.config

Commit:
Update webpack.config

Commit:
Add another external

Commit:
Merge pull request #2004 from mvines/de2

dialog-script now works on device

Commit:
:knife: :knife: :knife: npm-shrinkwrap.json

Commit:
0.17.2

Commit:
Move to node v4.4.5

Commit:
Remove unused external/npm

Commit:
Set SILK_DEVICE_MAIN_PATH more generically

Commit:
Set SILK_DEVICE_MAIN_PATH more generically

Commit:
Update nan to 2.3.5

lightsofapollo committed:
Merge pull request #1983 from lightsofapollo/webrtc-upgrade-android

Webrtc upgrade android

Commit:
Move to node v6.2.1

Commit:
Merge pull request #2011 from mvines/bump

automerge

Commit:
Update webpack.config

Commit:
0.18.0

Commit:
Merge pull request #2014 from mvines/node6

automerge

Commit:
<type_traits> in some gonks need help finding std::size_t

philikon committed:
Enable Flow for all files in cli

philikon committed:
Merge pull request #2040 from philikon/flowify_cli

Flowify cli-plugin-internal

Commit:
Triple camera init timeout to give slower sensors more leash

fangelathuge committed:
initial commit to enable Victara support

mvines commented:
This looks really nice!  victara.xml seems good to me except that I don't see where the kernel is coming from? I see a reference to `kernel/motorola/msm8974` [here](https://github.com/CyanogenMod/android_device_motorola_victara/blob/cm-12.1/BoardConfig.mk#L52), but I don't see how that gets added to victara.xml.

> ...Am I crazy or this should be enough to get an almost pure AOSP build for the device

That may work.  I'd suggest that you also confirm that the resulting AOSP build actually builds, boots, and works (IMO check out Camera, Wifi, BT, Sensors) before removing all the Android things and patching it into Silk to avoid wasting time debugging things on Silk that never worked on the original AOSP build to begin with.

--------
About the patches that you had questions about:

> 0001-Don-t-enable-MTP-by-default.patch

This one probably doesn't matter.   This was added to Mako because I found it annoying that my Linux machine always tried to connect with Mako over MTP every time it booted into Silk, even though we don't support MTP currently.

> 0001-Enable-firmware-to-mount-with-SELinux-disabled.patch 

It looks like this isn't needed for Victara as it's fstab.qcom has no SELinux context mount argument.  The symptom of not having this patch, if it's needed, would be that things like Wifi/BT/Camera won't work because the firmware for those subsystems can't be loaded from `/firmware`.

> 0001-Keep-2-cores-online-at-1.2ghz-to-stay-cool.patch

Yeah leave this one out too for now.  Mako has really poor thermal mitigation and will happily melt itself into a puddle if you work it too hard for too long, so we added this patch to keep it cool.  Kenzo is much better though, we're running that device with all 6 cores online and don't need this on that device either.

Commit:
Avoid flashing a black screen for a moment when switching movies

angelpipe commented:
Hey Michael,

As always thanks for your help. I decided to change my approach because cyanogen resources were giving me a lot of trouble during build. I decided to take a working XML from an Aurora build for victara and cleaned it (as it looks you did with kenzo). It means i had to look closer at patches and i rewrote some of them in the device tree to make them work. I'm having a final issue: the patch *system/core/0001-Permit-executables-in-system-node_modules.patch* inside android-5.1.1_r1 does not work in my current codebase, but it is a general patch imported for all devices; so i figured out there are 2 possible solutions to this.
1. Remove that patch from the android-5.1.1_r1 folder and add it on each device tree. so i can make a different one for victara.
2. Don't use android-5.1.1_r1.xml in victara.xml and copy all its patches into victara's tree.

Do you have any idea?

Best regards

mvines commented:
How about just placing a version of `0001-Permit-executables-in-system-node_modules.patch` that does apply to Victara at `bsp-gonk/board/victara/patch/system/core/0001-Permit-executables-in-system-node_modules.patch`.  Doing so will override override the patch in `bsp-gonk/board/android-5.1.1_r1/patch/system/core/0001-Permit-executables-in-system-node_modules.patch` that doesn't apply.   Does that work?

angelpipe commented:
sure. I didn't know that was possible. I will check and if all my tests go ok i think i will send a new PR and we should decline this one

andreasgal commented:
@angelpipe you can just push an update to this pull request

Commit:
Maintain a reference to the locked frame to avoid premature destruction

Commit:
NULL check properly

fangelathuge committed:
Change of approach. Aurora sources and new patches

angelpipe commented:
Hi, I just sent my changes. There's a new *victara.xml*. I used kenzo patches, fixed some of them that were not working and wrote new patches for device. Unfortunately i'm having issues with my environment and I'm not able to build anything(even mako) so i don't know if these changes could contain errors :(

Commit:
Follow symlinks when building node module dependencies

mvines commented on bsp-gonk/board/victara/patch/build/0001-Enable-ro.secure-0-for-userdebug.patch:
```diff
@@ -0,0 +1,39 @@
+From 1a17d44154df2dc5aceb6c01525542dfd689739c Mon Sep 17 00:00:00 2001
```
We probably don't need this patch but also pretty harmless to keep it, so meh

mvines commented on bsp-gonk/board/victara/patch/build/0004-Add-mechanism-to-disable-dependencies-for-the-otapac.patch:
```diff
@@ -0,0 +1,45 @@
+From 6b74dba481f079fd5e3713979b45c3f1df7db5b3 Mon Sep 17 00:00:00 2001
```
I vote to remove this patch.  It's really just a hack to avoid running out of disk space on Travis when building an OTA update package for the board.  We do this for our own purposes on Kenzo but Victara likely won't have this problem

mvines commented:
PR looks great.  When you figure out your env issues (:scream:) and are able to actually test it out a little, please let me know here and we can get this merged!

benturner committed:
Update webpack.config.js

Commit:
Merge pull request #2097 from benturner/dialog

automerge

fangelathuge committed:
removing unnecesary patch build/004

angelpipe commented:
Hey @mvines, i addressed the change you told me. About my environment, I'm on a Mac and i have got some problems to make it to work. I will continue trying to build, but I'm running out of ideas. The error I'm getting right now is:

```In file included from system/core/include/utils/String8.h:22,
                 from system/core/include/utils/CallStack.h:22,
                 from system/core/libutils/CallStack.cpp:19:
system/core/include/utils/Unicode.h:37: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:38: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:39: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:40: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:41: error: expected constructor, destructor, or type conversion before ‘*’ token
system/core/include/utils/Unicode.h:42: error: expected constructor, destructor, or type conversion before ‘*’ token
system/core/include/utils/Unicode.h:51: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:54: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:57: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:58: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:64: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:101: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:116: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:123: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/Unicode.h:153: error: ‘char32_t’ has not been declared
system/core/include/utils/Unicode.h:165: error: expected constructor, destructor, or type conversion before ‘*’ token
system/core/include/utils/Unicode.h:172: error: ‘char16_t’ has not been declared
system/core/include/utils/Unicode.h:179: error: expected constructor, destructor, or type conversion before ‘*’ token
In file included from system/core/include/utils/CallStack.h:22,
                 from system/core/libutils/CallStack.cpp:19:
system/core/include/utils/String8.h:55: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/String8.h:56: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/String8.h:56: error: ‘android::String8::String8(int)’ cannot be overloaded
system/core/include/utils/String8.h:55: error: with ‘android::String8::String8(int)’
system/core/include/utils/String8.h:57: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/String8.h:57: error: ‘android::String8::String8(int)’ cannot be overloaded
system/core/include/utils/String8.h:55: error: with ‘android::String8::String8(int)’
system/core/include/utils/String8.h:58: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/String8.h:58: error: ‘android::String8::String8(int)’ cannot be overloaded
system/core/include/utils/String8.h:55: error: with ‘android::String8::String8(int)’
system/core/include/utils/String8.h:79: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/String8.h:80: error: expected ‘,’ or ‘...’ before ‘*’ token
system/core/include/utils/String8.h:81: error: ‘android::status_t android::String8::setTo(int)’ cannot be overloaded
system/core/include/utils/String8.h:79: error: with ‘android::status_t android::String8::setTo(int)’
system/core/include/utils/String8.h:96: error: ‘char32_t’ has not been declared
make: *** [out/host/darwin-x86/obj32/STATIC_LIBRARIES/libutils_intermediates/CallStack.o] Error 1```

I know this is not your concern, but maybe you have seen it before.

Have a nice day!

mvines commented:
Hmm..do you have this patch applied?  I've run into Unicode.h issues before on osx, although don't immediately recognize this particular error.  https://github.com/silklabs/silk/blob/b65f76cee8bed4deea0cfb456f9e565fe4231a37/bsp-gonk/board/android-5.1.1_r1/patch/system/core/0001-OSX-Patches.patch

lightsofapollo committed:
Teach mocha to process unhandledRejections

angelpipe commented:
Yes, i have the patch applied. But this portion from Unicode.h looks suspicious

```#include <sys/types.h>
#include <stdint.h>
#ifdef __APPLE__
#include <new>
#endif

extern "C" {

// Definitions exist in C++11
#if defined __cplusplus && __cplusplus < 201103L
#ifndef __APPLE__
typedef uint32_t char32_t;
typedef uint16_t char16_t;
#endif
#endif```

Error shows up anytime the code tries to use *char_16_t* or *char_32_t*

lightsofapollo committed:
Merge pull request #2108 from lightsofapollo/unhandled

Teach mocha to process unhandledRejections

Commit:
Add support to play sound on host OSX and Ubuntu

Commit:
Add programs to dump mic data to stdout

Commit:
Reset errno properly

Commit:
Merge pull request #2112 from jainanshul/kws_test

automerge

Commit:
Fix M gonks

Commit:
Cleanup capture-mic dependencies

Commit:
TAG_MIC packet length in milliseconds is now divisible by 10

benturner committed:
Upgrade noble

benturner committed:
Merge pull request #666 from benturner/upgrade

Upgrade bleno and noble

Commit:
Update install

Commit:
Add null product

Commit:
Keep going if opencv3 is not present in the build

Commit:
Add github remote

Commit:
QEmu archives now support both Linux and Darwin

Commit:
Linux emulator tarball now supports Darwin too

angelpipe commented:
Hi,
After some digging, i've fixed some of the problems i faced with building but now im getting this one:

```
target thumb C++: silk-capture-daemon <= vendor/silk/capture/Capture.cpp
vendor/silk/bledroid/main.cpp:899:1: error: invalid conversion from 'void (*)(bt_bdaddr_t*, bt_bdname_t*, uint32_t) {aka void (*)(bt_bdaddr_t*, bt_bdname_t*, unsigned int)}' to 'pin_request_callback {aka void (*)(bt_bdaddr_t*, bt_bdname_t*, unsigned int, unsigned char)}' [-fpermissive]
 };
 ^
vendor/silk/bledroid/main.cpp:899:1: error: missing initializer for member 'bt_callbacks_t::le_lpp_write_rssi_thresh_cb' [-Werror=missing-field-initializers]
vendor/silk/bledroid/main.cpp:899:1: error: missing initializer for member 'bt_callbacks_t::le_lpp_read_rssi_thresh_cb' [-Werror=missing-field-initializers]
vendor/silk/bledroid/main.cpp:899:1: error: missing initializer for member 'bt_callbacks_t::le_lpp_enable_rssi_monitor_cb' [-Werror=missing-field-initializers]
vendor/silk/bledroid/main.cpp:899:1: error: missing initializer for member 'bt_callbacks_t::le_lpp_rssi_threshold_evt_cb' [-Werror=missing-field-initializers]
cc1plus: all warnings being treated as errors
make: *** [out/target/product/victara/obj/EXECUTABLES/silk-bledroid-daemon_intermediates/main.o] Error 1
make: *** Waiting for unfinished jobs....

#### make failed to build some targets (03:13 (mm:ss)) ####
```

Do you have any clue of what i'm doing wrong?

mvines commented:
Yes, this is very familiar.  The bluetooth HAL types tends to vary between different devices and SoCs, and we use build-time flags to manage this.

Victara seems to want [QBLUETOOTH_L](https://github.com/silklabs/silk/blob/bf4bd4acd/bsp-gonk/vendor/silk/bledroid/main.cpp#L890-L895), which can be defined by the make flag 
[TARGET_USES_QCOM_BSP](https://github.com/silklabs/silk/blob/bf4bd4acd/bsp-gonk/vendor/silk/bledroid/Android.mk#L8-L10).

Kenzo defines `TARGET_USES_QCOM_BSP := true` in it's device/qcom/msm8952_64/BoardConfig.mk (visible [here](https://github.com/silklabs/silk/blob/b65f76ce/bsp-gonk/board/kenzo/patch/device/qcom/msm8952_64/0001-BoardConfig-adjustments.patch#L33)), how about adding the same in `bsp-gonk/device/moto/victara/BoardConfig.mk`?

fangelathuge committed:
Build fix patches

Commit:
Move mic data into its own capture data channel

fangelathuge committed:
add silk product fix

fangelathuge committed:
adding generic goldfish device

Commit:
Merge pull request #2118 from mvines/mic

device: WebRTC audio recording support

Commit:
Add Nexus 5 (hammerhead) board

Commit:
Add Nexus 6 (shamu) board

Commit:
Add to v0.18.0 release notes

Commit:
Merge pull request #2129 from mvines/n56

Open source Nexus 5 and 6 boards

fangelathuge committed:
last round of patches to fix build

fangelathuge committed:
adding SElinux patch again

angelpipe commented:
Hi @mvines,

At the end i was able to build with the latest patches i have sent. Unfortunately, the build is not working good. I'm almost sure i'm having problems with SELinux, but i don't know what I'm doing wrong. I say problems are related to SELinux for this kind of errors:

```
--------- beginning of main
W/auditd  (  270): type=2000 audit(0.0:1): initialized
I/auditd  (  270): type=1403 audit(0.0:2): policy loaded auid=4294967295 ses=4294967295
I/chown   (  242): type=1400 audit(0.0:3): avc: denied { setattr } for uid=0 name="mmcblk0p20" dev="tmpfs" ino=10416 scontext=u:r:init:s0 tcontext=u:object_r:block_device:s0 tclass=blk_file permissive=1
I/ueventd (  258): type=1400 audit(0.0:4): avc: denied { search } for uid=0 name="/" dev="mmcblk0p1" ino=2 scontext=u:r:ueventd:s0 tcontext=u:object_r:unlabeled:s0 tclass=dir permissive=1
```
```
I/kmsg    (  387): <6>[   24.077935,0] pil-q6v5-mss fc880000.qcom,mss: mba: Brought out of reset
I/kmsg    (  387): <6>[   24.081561,0] pil-q6v5-mss fc880000.qcom,mss: modem: factory modem check
I/kmsg    (  387): <3>[   24.081747,0] pil-q6v5-mss fc880000.qcom,mss: modem: cert not found!
I/kmsg    (  387): <6>[   24.081905,0] pil-q6v5-mss fc880000.qcom,mss: modem: loading from 0x07700000 to 0x0d100000
I/ThermalEngine(  364): Sensor:chg_therm:32000 mC
I/kmsg    (  387): <3>[   24.468519,0] init: Warning!  Service silk needs a SELinux domain defined; please fix!
I/kmsg    (  387): <3>[   24.469098,0] init: Warning!  Service silk-bledroid needs a SELinux domain defined; please fix!
I/kmsg    (  387): <3>[   24.469559,0] init: Warning!  Service silk-sensors needs a SELinux domain defined; please fix!
E/QC-QMI  (  364): qmi_client [364]: unable to connect to server, errno=[2:No such file or directory], attempt=20
```
Maybe i translated wrong this patch https://github.com/angelpipe/silk/blob/pr/victara/bsp-gonk/board/victara/patch/device/moto/victara/0001-Add-silk-product.patch. But i tried without applying it too and I'm getting the same errors. Once again, do you ha a clue?

Thanks for your help!

mvines commented:
Nice, try adding `androidboot.selinux=permissive` to the victara's BOARD_KERNEL_CMDLINE.

```diff
diff --git a/BoardConfig.mk b/BoardConfig.mk
index 84a0b8e..0ef3e1a 100644
--- a/BoardConfig.mk
+++ b/BoardConfig.mk
@@ -39,7 +39,7 @@ TARGET_CPU_VARIANT := krait
 TARGET_USE_QCOM_BIONIC_OPTIMIZATION := true
 
 # Kernel
-BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.hardware=qcom msm_rtb.filter=0x37 ehci-hcd.park=3 vmalloc=
+BOARD_KERNEL_CMDLINE := androidboot.selinux=permissive console=ttyHSL0,115200,n8 androidboot.hardware=qcom msm_rtb.filte
 BOARD_KERNEL_BASE := 0x80200000
 BOARD_KERNEL_PAGESIZE := 2048
 BOARD_MKBOOTIMG_ARGS := --ramdisk_offset 0x02000000 --tags_offset 0x01e00000
```

angelpipe commented:
The file already had that parameter :(

mvines commented:
Hmm, I don't see too much else interesting from that log snippet unfortunately.  What does running `adb shell /system/silk/bin/silk-init` do?  If you put the full output of `adb logcat` right from boot in a gist, I'll scan it to see if I spot anything unusual.

angelpipe commented:
adb shell /system/silk/bin/silk-init:
```
H6460:victara fangel$ adb shell /system/silk/bin/silk-init
module.js:442
    throw err;
    ^

Error: Cannot find module 'silk-log/device'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/system/silk/node_modules/silk-sysutils/lib/index.js:423:19)
    at __webpack_require__ (/system/silk/node_modules/silk-sysutils/lib/index.js:21:30)
    at Object.<anonymous> (/system/silk/node_modules/silk-sysutils/lib/index.js:74:16)
    at __webpack_require__ (/system/silk/node_modules/silk-sysutils/lib/index.js:21:30)
    at module.exports.Object.defineProperty.value (/system/silk/node_modules/silk-sysutils/lib/index.js:41:18)
    at Object.<anonymous> (/system/silk/node_modules/silk-sysutils/lib/index.js:44:10)
```

The long gist you asked for :P --> https://gist.github.com/angelpipe/cc6417649b1eb7bf347d8eb02b093c25

mvines commented:
Thanks, so the logcat looks pretty good.  I see things from init.silk.rc starting up, like our Bluetooth daemon.  Silk-init is failing at boot in the same way as when it's run explicitly.  This no longer looks like a victara porting issue and a more general silk build issue, so \o/ for that 

For some reason, we're failing to require log/device at https://github.com/silklabs/silk/blob/master/bsp-gonk/vendor/silk/silk-sysutils/lib/index.js#L21.  This dependency should be resolved at build time by webpack.

It's possible there was a build error for silk-sysutils that didn't halt the build.  Could you try rebuilding sys-sysutils (run mm in https://github.com/silklabs/silk/tree/master/bsp-gonk/vendor/silk/silk-sysutils) to check for an error?  Otherwise the contents of /system/silk/node_modules/silk-sysutils/lib/index.js would be interesting to look at.  I'll double check later today that I'm not seeing the same in a Nexus build .

angelpipe commented:
It's good to hear logcat is not plaged with errors :). I have run mm and it looks to run ok, this is the result:
```
H6460:silk-sysutils fangel$ mm
============================================
PLATFORM_VERSION_CODENAME=REL
PLATFORM_VERSION=5.1.1
TARGET_PRODUCT=aosp_victara
TARGET_BUILD_VARIANT=userdebug
TARGET_BUILD_TYPE=release
TARGET_BUILD_APPS=
TARGET_ARCH=arm
TARGET_ARCH_VARIANT=armv7-a-neon
TARGET_CPU_VARIANT=krait
TARGET_2ND_ARCH=
TARGET_2ND_ARCH_VARIANT=
TARGET_2ND_CPU_VARIANT=
HOST_ARCH=x86_64
HOST_OS=darwin
HOST_OS_EXTRA=Darwin-14.5.0-x86_64-i386-64bit
HOST_BUILD_TYPE=release
BUILD_ID=LMY47V
OUT_DIR=out
============================================
SILK_PRODUCT: gonkjs
SILK_BOARD: victara
============================================
No private recovery resources for TARGET_DEVICE victara
Export includes file: vendor/silk/silk-sysutils/Android.mk -- out/target/product/victara/obj/NPM/silk-sysutils_intermediates/export_includes

#### make completed successfully (7 seconds) ####
```
About /system/silk/node_modules/silk-sysutils/lib/index.js, i created a new gist so you can check it here:
https://gist.github.com/angelpipe/20021477792c0d78a977a576c3a40bac

I hope we are really close to have a working port.

Commit:
Ditch the long list of APNs

Commit:
Remove stale osx gcc from N[4-6] manifests

Commit:
Fix up blob path

Commit:
Correct silk-log symlink

Commit:
Movie.run() expects a callback

Commit:
Add process error handlers, null movie callback

mvines commented:
Thanks, the contents of index.js was very helpful.  We had a bug, now fixed by a813116d7f9cacb59e70a00bfe53ea99aec40a39, whereby webpack was unable to locate the `silk-log/device` module at build time.  

Please rebase the PR to the tip of the `master` branch and rebuild.  Hopefully Victara now boots and we can land this PR!

fangelathuge committed:
last change before rebasing

angelpipe commented:
I don't know if this is good news, your fix worked!. The problem now is im creating a simple silk project and doing silk run. Im getting this
```
H6460:test fangel$ silk run
Updating /data/node_modules/test
--------- beginning of main
06-19 19:58:37.379 I/silk-init( 4716): Starting Silk 
06-19 19:58:37.896 I/silk-init( 4683): initializing silk with main: test
06-19 19:58:38.039 I/silk-wifi( 4686): (+0ms) WiFi initializing
06-19 19:58:38.404 I/silk-wifi( 4686): (+366ms) wlan0 state| 213 0 00:00:00:00:00:00 0.0.0.0 0 down
06-19 19:58:38.404 I/silk-wifi( 4686): 
06-19 19:58:38.413 I/silk-wifi( 4686): (+10ms) wpaCli: supplicant not ready, waiting
06-19 19:58:38.423 I/silk-wifi( 4686): (+9ms) wpad restart: wpad error, reason=Error: connect ECONNREFUSED /dev/socket/wpad
06-19 19:58:39.420 I/silk-wifi( 4686): (+997ms) wpaCli: supplicant not ready, waiting
06-19 19:58:39.435 I/silk-wifi( 4686): (+15ms) wpad restart: wpad error, reason=Error: connect ENOENT /dev/socket/wpad
```
Could it be related to a missing driver?

jainanshul commented:
@angelpipe this is due to a race condition where silk is trying to start wpa before it is ready. This usually resolves itself in couple of seconds when wpa subsystem is ready.

angelpipe commented:
Hi @jainanshul. The problem is it keeps showing that message over and over until i stop it. I could leave it running for half an hour and the message keeps printing :(

angelpipe commented:
i found this on device logcat, i think there's a problem with the wpa subsystem and it is my fault
```
I/wpa_supplicant(  564): Successfully initialized wpa_supplicant
I/kmsg    (  364): <38>[    4.621397,0] type=1400 audit(1466380924.772:50): avc: denied { execmem } for uid=0 pid=447 comm="node" scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=process permissive=1
E/wpa_supplicant(  564): Line 1: unknown global field 'mot_wpa_conf_version=3'.
E/wpa_supplicant(  564): Line 1: Invalid configuration line 'mot_wpa_conf_version=3'.
E/wpa_supplicant(  564): Line 25: unknown network field 'skip_auto_connect'.
E/wpa_supplicant(  564): Line 26: failed to parse network block.
I/wpa_supplicant(  564): Long line in configuration file truncated
I/wpa_supplicant(  564): rfkill: Cannot open RFKILL control device
E/wpa_supplicant(  564): Could not read interface wlan0 flags: No such device
E/wpa_supplicant(  564): Could not read interface wlan0 flags: No such device
E/wpa_supplicant(  564): wlan0: Failed to initialize driver interface
```

mvines commented:
Oh, yep that looks suspicious!

I'm not sure if this is still the case, but on the previous device logcat I also noticed this which looked like the modem was not happy -- https://gist.github.com/angelpipe/cc6417649b1eb7bf347d8eb02b093c25#file-logcat-L2643-L2665
Generally we don't care too much about that because we don't have telephony support in Silk yet but that could potentially be a symptom of a larger issue.

mvines commented:
@angelpipe -- hey are you still working on anything here, just wondering if this is PR ready to merge.  Thanks!

angelpipe commented:
Well.... i never stopped working, but i haven't been able to make the wpa_suplicant work. Right now i'm trying a new build structure

angelpipe commented:
Hey @mvines. I haven't had luck nor fixing the wpa_supplicant problem nor trying to build with different sources. I don't know if i should spend more time trying to fix it or if i should try getting something that proved to work (like a CyanogenMod complete manifest) and migrate all patches to it to then remove the unnecessary things. Motorola has some info to build android for a variety of devices but sadly, Moto X 2014 is not on the instructions and repos look outdated. What would you recommend to do?

mvines commented:
Ugh. In my experience it's super helpful to have a buidable Android image to start from that demonstrates all the desired features (eg, wifi works) . Given the amount of vendor variance in Android I'd not want to start with anything less, there's to much risk of running into a binary blob that does something different with very little way to debug/trace it

angelpipe commented:
Hey... I built CyanogenMod 12.1 for victara, it works good when i do it following instructions. Then i tried to integrate my manifest into the bsp-gonk board scheme and built. I had some errors and i managed to fix them to get a successful build. The device keeps booting in the "android" screen. It even does not show the cyanogen logo(which i think is weird as it comes from a cyanogen manifest). I cannot check the logcat cause cyanogenmod comes with that disabled (Maybe i can tinker to make it work from scratch?). The thing is when i build the Silk style, i don't get the cyanogen zip, i just get the images and the generated recovery does not work. Maybe i am flashing it the wrong way?. This way is not proving to be easier :(. Do you have any thoughts?
